src/item.cpp

```diff
@@ -2499,6 +2500,27 @@ int item::precise_unit_volume() const
     return volume() * 1000;
 }
 
+int item::base_volume() const
+{
+    if( is_null() ) {
+        return 0;
+    }
+
+    if( is_corpse() ) {
+        switch( corpse->size ) {
+            case MS_TINY:    return    3;
+            case MS_SMALL:   return  120;
+            case MS_MEDIUM:  return  250;
+            case MS_LARGE:   return  370;
+            case MS_HUGE:    return 3500;
+        }
+        debugmsg( "unknown monster size for corpse" );
+        return 0;
+    }
+
+    return type->volume;
+}
+
 int item::volume( bool integral ) const
 {
     if( is_null() ) {
```

Comment:
Since `item::base_volume()` duplicates a chunk of code from the `item::volume()` the latter should call the former.